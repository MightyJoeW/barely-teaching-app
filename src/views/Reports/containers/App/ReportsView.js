import React, { Component } from "react";
import { BrowserRouter, Route, Switch, Link, Redirect } from "react-router-dom";

import AddReportView from "../AddReportView/AddReportView";
import EditReportView from "../EditReportView/EditReportView";
import ReportsHeader from "../ReportsHeader/ReportsHeader";
import ReportsNotFoundView from "../ReportsNotFoundView/ReportsNotFoundView";
import ReportList from "../../components/ReportList";
import ReportListFilters from "../../components/ReportListFilters";

export default class Reports extends Component {
  render() {
    return (
      <div>
        <BrowserRouter>
          <div>
            {
              // <ReportList />
            }
            <Switch>
              <Route exact path="/reports" component={ReportList} />
              <Route path="/reports/create" component={AddReportView} />
              <Route path="/reports/edit/:id" component={EditReportView} />
              {/* <Route component={ReportsNotFoundView} /> */}
            </Switch>
          </div>
        </BrowserRouter>
      </div>
    );
  }
}
