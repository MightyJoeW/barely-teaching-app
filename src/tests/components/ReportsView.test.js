import React from "react";
import { shallow } from "enzyme";
import ReportsView from "../../views/Reports/containers/App/ReportsView";

test("should render ReportsView correctly", () => {
  const wrapper = shallow(<ReportsView />);
  expect(wrapper).toMatchSnapshot();
});
