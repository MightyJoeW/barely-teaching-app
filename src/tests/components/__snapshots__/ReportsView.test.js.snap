// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ReportsView correctly 1`] = `
<div>
  <BrowserRouter>
    <div>
      <ReportsHeader />
      <Connect(ReportListFilters) />
      <Connect(ReportList) />
      <Switch>
        <Route
          component={[Function]}
          path="/reports/create"
        />
        <Route
          component={[Function]}
          path="/reports/edit/:id"
        />
      </Switch>
    </div>
  </BrowserRouter>
</div>
`;
